packages/ui/primitives/pin-input.tsx

```diff
@@ -0,0 +1,85 @@
+'use client';
+
+import { useRef } from 'react';
+
+import { CodeInput, getSegmentCssWidth } from 'rci';
+import { useIsFocused } from 'use-is-focused';
+
+import { cn } from '@documenso/ui/lib/utils';
+
+export type PinInputState = 'input' | 'loading' | 'error' | 'success';
+export type PinInputProps = {
+  id: string;
+  state: PinInputState;
+  autoFocus?: boolean;
+  onSubmit({ code, input }: { code: string; input: EventTarget & HTMLInputElement }): void;
+};
+
+const PinInput = ({ id, autoFocus, state, onSubmit }: PinInputProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const focused = useIsFocused(inputRef);
+
+  const width = getSegmentCssWidth('14px');
+
+  return (
+    <CodeInput
+      id={id}
+      className={cn({
+        'motion-safe:animate-[shake_0.15s_ease-in-out_0s_2]': state === 'error',
+      })}
+      inputClassName="caret-transparent selection:bg-transparent ring:ring-2"
+      autoFocus={autoFocus}
+      length={6}
+      fontFamily="Inter"
```

Comment:
You will want a monospaced font for this otherwise you'll get the funky sizing as per your video. I assume you can skip a lot of this prop drilling though and just add it to `inputClassName`?

Reply:
I'll do that